app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +76,4826 @@ const PRE_1559_BLOCK = {
+            expect(controller.store.getState().network).toBe('loading');
+          },
+        );
+      });
+
+      it('resets EIP support for the network before emitting networkDidChange', async () => {
+        await withController(
+          {
+            state: {
+              provider: {
+                type: 'rpc',
+                rpcUrl: 'https://mock-rpc-url',
+                chainId: '0xFF',
+              },
+            },
+          },
+          async ({ controller, network }) => {
+            network.mockEssentialRpcCalls({
+              latestBlock: POST_1559_BLOCK,
+              eth_blockNumber: {
+                times: 2,
+              },
+            });
+
+            await controller.initializeProvider();
+            expect(controller.store.getState().networkDetails).toStrictEqual({
+              EIPS: {
+                1559: true,
+              },
+            });
+
+            await waitForStateChanges({
+              controller,
+              propertyPath: ['networkDetails'],
+              // We only care about the first state change, because it happens
+              // before networkDidChange
+              count: 1,
+              operation: () => {
+                controller.resetConnection();
+              },
+            });
+            expect(controller.store.getState().networkDetails).toStrictEqual({
+              EIPS: {
+                1559: undefined,
+              },
+            });
+          },
+        );
+      });
+
+      it('initializes a new provider object pointed to the same RPC URL as the current network and using the same chain ID', async () => {
+        await withController(
+          {
+            state: {
+              provider: {
+                type: 'rpc',
+                rpcUrl: 'https://mock-rpc-url',
+                chainId: '0x1337',
+              },
+            },
+          },
+          async ({ controller, network }) => {
+            network.mockEssentialRpcCalls();
+
+            controller.resetConnection();
+
+            const { provider } = controller.getProviderAndBlockTracker();
+            const promisifiedSendAsync = promisify(provider.sendAsync).bind(
+              provider,
+            );
+            const { result: chainIdResult } = await promisifiedSendAsync({
+              method: 'eth_chainId',
+            });
+            expect(chainIdResult).toBe('0x1337');
+          },
+        );
+      });
+
+      it('replaces the provider object underlying the provider proxy without creating a new instance of the proxy itself', async () => {
+        await withController(
+          {
+            state: {
+              provider: {
+                type: 'rpc',
+                rpcUrl: 'https://mock-rpc-url',
+                chainId: '0x1337',
+              },
+            },
+          },
+          async ({ controller, network }) => {
+            network.mockEssentialRpcCalls();
+            controller.initializeProvider();
```

Comment:
I noiced that the provider is being initialized above without an `await`. There are five tests total that call `initializeProvider` without `await`. Was this intentional?

Reply:
Hmm, I thought I fixed these :( That is not intentional. That might explain some of the waiting that I have to do in later tests. Fixed in [1ed0356](https://github.com/MetaMask/metamask-extension/pull/17498/commits/1ed03565ede7825bd5797fee5ba9984caaa5f3d9).